cpp/tests/test_abm_world.cpp

```diff
@@ -589,135 +589,251 @@ TEST(TestWorld, copyWorld)
-    ASSERT_NE(&(copied_world.get_persons()[0].get_cells()), &world.get_persons()[0].get_cells());
-    ASSERT_NE(&(copied_world.get_persons()[1].get_cells()), &world.get_persons()[1].get_cells());
+    EXPECT_NE(&(copied_world.get_persons()[0].get_infection()), &world.get_persons()[0].get_infection());
+    EXPECT_NE(&(copied_world.get_persons()[0].get_mask()), &world.get_persons()[0].get_mask());
+    EXPECT_NE(&(copied_world.get_persons()[1].get_mask()), &world.get_persons()[1].get_mask());
+    EXPECT_NE(&(copied_world.get_persons()[0].get_cells()), &world.get_persons()[0].get_cells());
+    EXPECT_NE(&(copied_world.get_persons()[1].get_cells()), &world.get_persons()[1].get_cells());
 
     // Evolve the world and check that the copied world has not evolved
     copied_world.get_persons()[0].migrate_to(work, {0});
     copied_world.get_persons()[1].migrate_to(home, {0});
-    ASSERT_NE(copied_world.get_persons()[0].get_location().get_type(),
+    EXPECT_NE(copied_world.get_persons()[0].get_location().get_type(),
               world.get_persons()[0].get_location().get_type());
-    ASSERT_NE(copied_world.get_persons()[1].get_location().get_type(),
+    EXPECT_NE(copied_world.get_persons()[1].get_location().get_type(),
               world.get_persons()[1].get_location().get_type());
 }
+
+TEST(TestWorld, migrateWithAppliedNPIs)
+{
+    using testing::Return;
+    // Test when the NPIs are applied, people can enter targeted location if they comply to the rules.
+    auto t     = mio::abm::TimePoint(0) + mio::abm::hours(8);
+    auto dt    = mio::abm::hours(1);
+    auto world = mio::abm::World(num_age_groups);
+    auto rng   = mio::RandomNumberGenerator();
+    world.parameters
+        .get<mio::abm::InfectedNoSymptomsToSymptoms>()[{mio::abm::VirusVariant::Wildtype, age_group_15_to_34}] =
+        2 * dt.days();
+    world.parameters.get<mio::abm::AgeGroupGotoWork>().set_multiple({age_group_15_to_34, age_group_35_to_59}, true);
+
+    auto home_id = world.add_location(mio::abm::LocationType::Home);
+    auto work_id = world.add_location(mio::abm::LocationType::Work);
+
+    ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>> mock_uniform_dist;
+    EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
+        .Times(testing::AtLeast(8))
+        .WillOnce(testing::Return(0.8)) // draw random work group
+        .WillOnce(testing::Return(0.8)) // draw random school group
+        .WillOnce(testing::Return(0.8)) // draw random work hour
+        .WillOnce(testing::Return(0.8)) // draw random school hour
+        .WillOnce(testing::Return(0.8)) // draw random work group
+        .WillOnce(testing::Return(0.8)) // draw random school group
+        .WillOnce(testing::Return(0.8)) // draw random work hour
+        .WillOnce(testing::Return(0.8)) // draw random school hour
+        .WillRepeatedly(testing::Return(1.0));
+
+    auto& p_complied =
```

Comment:
```suggestion
    auto& p_compliant =
```
Suggesting to call it like this, also in other places.

Reply:
It's fixed!